invenio_jobs/jobs.py

```diff
@@ -39,17 +45,29 @@ def create(
             ),
         )
 
-    @abstractmethod
-    def default_args(self, *args, **kwargs):
-        """Abstract method to enforce implementing default arguments."""
+    @classmethod
+    def get_task_args(cls, job_obj, **kwargs):
+        """Override to define extra arguments to be injected on task execution.
+
+        :param job_obj (Job): the Job object.
+        :return: a dict of arguments to be injected on task execution.
+        """
         return {}
 
     @classmethod
-    def build_task_arguments(cls, job_obj, custom_args=None, **kwargs):
-        """Build arguments to be passed to the task.
+    def build_task_arguments(cls, job_obj, since=None, custom_args=None, **kwargs):
```

Comment:
This should probably be private, used only internally in this module:
```suggestion
    def _build_task_arguments(cls, job_obj, since=None, custom_args=None, **kwargs):
```